packages/website/src/pages/index.tsx

```diff
@@ -108,7 +108,7 @@ const features: FeatureItem[] = [
 function Feature({ title, description }: FeatureItem): JSX.Element {
   return (
     <div className="col col--12 padding-vert--lg">
-      <h2 className="text--center">{title}</h2>
+      <h2 id="title" className="text--center">{title}</h2>
```

Comment:
Docusaurus has a `Heading` component. In order for the clickable "#" to show up you have to use that. This isn't documented, but check out https://github.com/facebook/docusaurus/blob/b4c04f28f760cb7ea9de709643df4c23f8b0e934/website/src/pages/versions.tsx#L64